unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/AssetManager/Font/Providers/FontAssetResolver.cs

```diff
@@ -0,0 +1,67 @@
+using Cysharp.Threading.Tasks;
+using MainScripts.DCL.Controllers.AssetManager;
+using MainScripts.DCL.Controllers.AssetManager.Font;
+using MainScripts.DCL.Controllers.AssetManager.Texture;
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using TMPro;
+using UnityEngine;
+
+namespace DCL
+{
+    public class FontAssetResolver : AssetResolverBase, IFontAssetResolver
+    {
+
+        private readonly IReadOnlyDictionary<AssetSource, IFontAssetProvider> providers;
+
+        public FontAssetResolver(IReadOnlyDictionary<AssetSource, IFontAssetProvider> providers, DataStore_FeatureFlag featureFlags)
+            : base(featureFlags)
+        {
+            this.providers = providers;
+        }
+
+        public async UniTask<FontResponse> GetFontAsync(AssetSource permittedSources, string url, CancellationToken cancellationToken = default)
+        {
+             Exception lastException = null;
+             TMP_FontAsset font = null;
+
+
+            using var permittedSourcesRent = GetPermittedProviders(this.providers, permittedSources);
+            var permittedProviders = permittedSourcesRent.GetList();
+
+            foreach (var provider in permittedProviders)
+            {
+                try
+                {
+                    font = await provider.GetFontAsync(url, cancellationToken);
+
+                    if (font)
+                    {
+                        // The valid font is retrieved
+                        AssetResolverLogger.LogVerbose(featureFlags, LogType.Log, $"Texture {url} loaded from {provider}");
```

Comment:
Logs refer to `Texture` 😄

Reply:
Took a little too much inspiration :sweat-smile:. Corrected!